backend/typescript/rest/animalTypeRoutes.ts

```diff
@@ -0,0 +1,103 @@
+import { Router } from "express";
+// import { isAuthorizedByRole } from "../middlewares/auth";
+import { animalTypeRequestDtoValidator } from "../middlewares/validators/animalTypeValidators";
+import AnimalTypeService from "../services/implementations/animalTypeService";
+import {
+  AnimalTypeResponseDTO,
+  IAnimalTypeService,
+} from "../services/interfaces/animalTypeService";
+import { getErrorMessage, NotFoundError } from "../utilities/errorUtils";
+import { sendResponseByMimeType } from "../utilities/responseUtil";
+
+const animalTypeRouter: Router = Router();
+// animalTypeRouter.use(isAuthorizedByRole(new Set(["User", "Admin"])));
+
+const animalTypeService: IAnimalTypeService = new AnimalTypeService();
+
+/* Create AnimalType */
+animalTypeRouter.post("/", animalTypeRequestDtoValidator, async (req, res) => {
+  try {
+    const { body } = req;
+    const newAnimalType = await animalTypeService.createAnimalType({
+      animal_type_name: body.animal_type_name,
+    });
+    res.status(201).json(newAnimalType);
+  } catch (e: unknown) {
+    res.status(500).send("Internal server error occured.");
+  }
+});
+
+/* Get all AnimalTypes */
+animalTypeRouter.get("/", async (req, res) => {
+  const contentType = req.headers["content-type"];
+  try {
+    const animalTypes = await animalTypeService.getAnimalTypes();
+    await sendResponseByMimeType<AnimalTypeResponseDTO>(
+      res,
+      200,
+      contentType,
+      animalTypes,
+    );
+  } catch (e: unknown) {
+    await sendResponseByMimeType(res, 500, contentType, [
+      {
+        error: "Internal server error occured.",
+      },
+    ]);
+  }
+});
+
+/* Get AnimalType by id */
+animalTypeRouter.get("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
+    const animalType = await animalTypeService.getAnimalType(String(id));
```

Comment:
When you get id from the request it should already be a string. I don't think you need to explicitly cast it?